NitroxTest/Serialization/WorldPersistenceTest.cs

```diff
@@ -0,0 +1,426 @@
+                {
+                    CompletedBasePieceHistory = new List<BasePiece>()
+                    {
+                            new BasePiece(new NitroxId(), NitroxVector3.Zero, NitroxQuaternion.Identity, NitroxVector3.Zero, NitroxQuaternion.Identity, new NitroxTechType("BasePiece1"), Optional.Empty, false, Optional.Empty)
+                    },
+                    PartiallyConstructedPieces = new List<BasePiece>()
+                    {
+                        new BasePiece(new NitroxId(), NitroxVector3.One, NitroxQuaternion.Identity, NitroxVector3.One, NitroxQuaternion.Identity, new NitroxTechType("BasePiece2"), Optional.Empty, false, Optional.Empty)
+                    }
+                },
+                EntityData = new EntityData()
+                {
+                    Entities = new List<Entity>()
+                    {
+                        new Entity(NitroxVector3.Zero, NitroxQuaternion.Identity, NitroxVector3.One, new NitroxTechType("Peeper"), 1, "PeeperClass", false, new NitroxId(), new byte[]{ 0x10, 0x14, 0x0, 0x2, 0x2, 0x2, 0x2 }, false, new NitroxId()),
+                        new Entity(NitroxVector3.One, NitroxQuaternion.Identity, NitroxVector3.One, new NitroxTechType("Peeper"), 1, "PeeperClass", false, new NitroxId(), new byte[]{ 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20 }, true, new NitroxId())
+                    }
+                },
+                PlayerData = new PlayerData()
+                {
+                    Players = new List<PersistedPlayerData>()
+                    {
+                        new PersistedPlayerData()
+                        {
+                            NitroxId = new NitroxId(),
+                            Id = 1,
+                            Name = "Test1",
+                            IsPermaDeath = false,
+                            Permissions = Perms.ADMIN,
+                            SpawnPosition = NitroxVector3.Zero,
+                            SubRootId = new NitroxId(),
+                            CurrentStats = new PlayerStatsData(45, 45, 40, 39, 28, 1),
+                            EquippedItems = new List<EquippedItemData>(0),
+                            Modules = new List<EquippedItemData>(0)
+                        },
+                        new PersistedPlayerData()
+                        {
+                            NitroxId = new NitroxId(),
+                            Id = 2,
+                            Name = "Test2",
+                            IsPermaDeath = false,
+                            Permissions = Perms.PLAYER,
+                            SpawnPosition = NitroxVector3.One,
+                            SubRootId = new NitroxId(),
+                            CurrentStats = new PlayerStatsData(40, 40, 30, 29, 28, 0),
+                            EquippedItems = new List<EquippedItemData>(0),
+                            Modules = new List<EquippedItemData>(0)
+                        }
```

Comment:
Is it worth adding another with permadeath as true and maybe some modules/equipteditems to make sure that they save correctly